In [78]:
"""
stuff you will find useful dealing with antennas
"""
import math
# some constants
c=2.998e8 # velocity of light in m/s
k=1.38e-23 # Boltzmann constant in J/K
Jy=1e-26 #W/m**2/Hz


# Math

$\lambda = \frac{c}{\nu}$

$A=\pi \frac{D^2}{4}$

$\theta = \frac{\lambda}{B}$

# For a single dish

$SEFD = \frac{T_{sys}}{K}$
where
$K=\frac{A \eta_A}{2k}$

for dish area A and efficiency $\eta_A$

##It is convenient to  convert SEFD to Jy

# For an array of N of the same dishes
USABLE observing bandwidth $\Delta\nu$

integration time $\tau$

system efficiency $\eta_s$

$\Delta I = \frac{SEFD} { \eta_s \sqrt{N(N-1)~ \tau ~\Delta\nu}}$



In [79]:
"""
More general stuff you will need for MeerKAT
"""

D=13.5 # dish diameter D


#Some estimates from Isak#

<table> 
<tr> <th>Subsystem</th> <th>900MHz</th> <th>1600MHz</th> </tr>
<tr> <td>Sky (zenith pointing) </td> <td> 6.6K</td> <td> 4.9K</td><tr> 
<tr> <td>Dish (transmission for a panelled dish)  </td> <td> 1.0K</td> <td> 1.0K</td><tr> 
<tr> <td>Receiver </td> <td> 7.2K</td> <td> 8.2K</td><tr> 
<tr> <td>Digitizer </td> <td> 0.5K</td> <td> 0.5K</td><tr> 
<tr> <td>TReceiver + TSky + TDigitiser </td> <td> 15.3K</td> <td> 14.6K</td><tr> 

</table>

# Assumptions
* All dishes have same Tsys (15K)
* All dishes have same efficiency $\eta_A$ = 0.75
* System efficiency  $\eta_s$ = 0.95 

In [83]:
Tsys=15
e_A=0.75
A = math.pi*D*D/4
K=A*e_A/(2*k)
SEFD=Tsys/(K*Jy) # converted to Jy
print "SEFD %.2f Jy" %(SEFD)

print "per dish A_e/Tsys %.1f" %(A*e_A/Tsys)

SEFD 385.64 Jy
per dish A_e/Tsys 7.2


In [84]:
"""
Array part
"""
N=64 # dishes
e_s= 0.95
BW=400e9 # bandwidth in Hz -Just the usable bit after flagging
t=60 # time in seconds
dI = SEFD /(e_s*math.sqrt(N*(N-1)*BW *t))
print "Noise %.2g JY" % (dI)

print "A_e/Tsys %.0f" %(e_s*N*A*e_A/Tsys)


Noise 1.3e-06 JY
A_e/Tsys 435


In [85]:
""""
Resolution part-  depends on wavelength and baseline lengths.

Note: various fudges here
- bandwidth is high, so there is no one wavelength
- real mapping involves choices of weighting function etc
"""
f =  1420e6 #Hz
lam= c/f   #m
Bmax= 4000 #m
Bmin= 20 #m
Btyp=Bmax/2  # wild approximation
resolution  = lam/Btyp #radians
#print lam, Btyp, resolution
resol = 60*60*math.degrees(resolution)
print "resolution %.0f arcseconds" %(resol)
print "sensitivity %.2g Jy/square arcsec" %(dI/(resol*resol))

resolution 22 arcseconds
sensitivity 2.8e-09 Jy/square arcsec


# Rayleigh-Jeans law
## valid for radio regime - not optical

$B_{\nu} = \frac{2 k T \nu^2}{c^2}$


$T =  \frac{ B_{\nu} c^2}{2 k \nu^2}$


In [88]:
#Now to convert to brightness sensitivty for the resolution we have at that resolution
# this is ABOVE the smooth background 
flx = dI * Jy # to W/m**2Hz
sterad=math.pi*resolution**2/4.0
B=flx/sterad

T=B*c*c/(2*k*f*f)
print "Sensitivity %.2g K" %(T)



Sensitivity 0.0024 K


# Formulae from papers:

$T= 1.3\times 10^6 (1+z) \frac{B}{\theta^2 \nu^2}$
where
* $\theta$ is angular size in arcsec
* S is flux density
* $\nu$ is freq in GHz

## various other authors 
give 1.22 to 1.8 for the inital constant
